### ISSUE TITLE
[Accessibility]: Ref panel is difficult to use on mobile

### ISSUE THREAD

**Issue by varungandhi-src** [sourcegraph/sourcegraph#36639]
### Audit type

Mobile device navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33510

### Problem description

I'm making this as one issue for now since these factors seem related.

1. There is a high degree of visual overlap between different elements, with different elements being clipped, flowing across multiple lines and so on. It seems like there is a more fundamental issue here, not just misuse of components or something, because the current design relies on using horizontal space for long paths and the file preview on the side, but horizontal space on mobile is very limited.
2. The Close code view button cannot be clicked; if I double-click on it, the "your reference panel feedback!" line gets selected.
3. Navigating and tapping in the code preview on the side is very cumbersome because of limited horizontal space.

### Expected behavior

1. Different elements shouldn't visually overlap or be clipped.
2. There should be more space available to the file preview.
3. All buttons should be clickable.

### Additional details

![](https://user-images.githubusercontent.com/93103176/172142188-7fcfb601-9a15-45bc-81e1-d4654d71f41b.png)

![](https://user-images.githubusercontent.com/93103176/172142997-c59dbb34-d2ec-468c-9eba-acbdf41ce3e1.png)

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [ ] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
